v2/emailpassword/common-customizations/email-verification/protecting-routes.mdx

```diff
@@ -1207,15 +1210,180 @@ async function checkIfEmailIsVerified() {
 </TabItem>
 
 <TabItem value="angular">
+<AngularUIImplementation>
+
+<Answer title="Prebuilt UI">
+
+```tsx
+import Session from "supertokens-auth-react/recipe/session";
+import { EmailVerificationClaim } from "supertokens-auth-react/recipe/emailverification";
+
+async function shouldLoadRoute(): Promise<boolean> {
+    if (await Session.doesSessionExist()) {
+
+        // highlight-start
+        let validationErrors = await Session.validateClaims();
+
+        if (validationErrors.length === 0) {
+            // user has verified their email address
+            return true;
+        }
+        // highlight-end
+    }
+    // either a session does not exist, or one of the validators failed.
+    // so we do not allow access to this page.
+    return false
+}
+```
+
+We call the `validateClaims` function. If you initialized the `EmailVerification` recipe, this will include the `EmailVerificationClaim` validator in the `globalValidators`.
+
+<details><summary>Checking validation error</summary>
+
+In case the `validationErrors` array is not empty, you can loop through the errors to know which claim has failed:
+
+```tsx
+import Session from "supertokens-auth-react/recipe/session";
+import { EmailVerificationClaim } from "supertokens-auth-react/recipe/emailverification";
+
+async function shouldLoadRoute() {
+    let validationErrors = await Session.validateClaims(/*{...}*/);
+    // highlight-start
+    for (const err of validationErrors) {
+        if (err.validatorId === EmailVerificationClaim.id) {
+            // email verification claim check failed
+        } else {
+            // some other claim check failed (from the global validators list)
+        }
+    }
+    // highlight-end
+}
+```
+
+</details>
+
+<details><summary>Manually checking verification status</summary>
+If you want to have more complex access control, you can either create your own validator, or you can get the boolean from the session as follows, and check it yourself:
+
+```tsx
+import Session from "supertokens-auth-react/recipe/session";
+import { EmailVerificationClaim } from "supertokens-auth-react/recipe/emailverification";
+
+async function shouldLoadRoute(): Promise<boolean> {
+    if (await Session.doesSessionExist()) {
+        // highlight-start
+        let isVerified = await Session.getClaimValue({claim: EmailVerificationClaim});
+        if (isVerified) {
+            // user has verified their email address
+            return true;
+        }
+        // highlight-end
+    }
+    // either a session does not exist, or the user has not verified their email address
+    return false
+}
+```
+
+</details>
+
+</Answer>
+
+<Answer title="Custom UI">
 
 ~COPY-TABS=npm,vanillajs
 
+</Answer>
+
+</AngularUIImplementation>
 </TabItem>
 
 <TabItem value="vue">
 
-~COPY-TABS=angular
+<VueUIImplementation>
+
+<Answer title="Prebuilt UI">
+
```

Comment:
why can't we use copy tabs here and copy from angular (for vue js - instead of copying the code manually)?